lib/pka_dev.c

```diff
@@ -1687,6 +1687,37 @@ bool pka_dev_has_trng(pka_dev_shim_t *shim)
     return (shim->trng_enabled == PKA_SHIM_TRNG_ENABLED);
 }
 
+int pka_dev_shim_get_random_bytes(uint32_t *data, uint32_t cnt)

```

Comment:
I think there is an issue here...

The idea is to isolate PKA resources from an instance to an other. We cannot loop over all shims since not all shims are dedicated to an instance. I suggest to have the instance loop over its available rings and figure out what shim could be used for random bytes generation. Think about two instances where the first instance owns rings within shim 1 and shim 2, and a second instance that owns rings within shim 3 and shim 4. We don't won't the first instance to interfere with the resources of the second instance. But when all shims are shared among instances then in this case I am assuming it is fine.